### ISSUE TITLE
[UX] `dvc remove` safety

### ISSUE THREAD

**Issue by colllin** [iterative/dvc#1524]
Backstory: I just `dvc remove`d 2 tarballs which took me several hours to generate 😭.

Suggestions: 
1. The current syntax of `dvc remove some.tgz.dvc` makes the user think they're removing the dvc file, which would seem to have some sort of "unlinking" functionality.  Instead, it leaves the `.dvc` file and deletes the file which is under dvc control.  This is unintuitive and destructive — a nasty combination.  If I wanted to delete that file, why wouldn't I just `rm some.tgz`?  What, if anything, did dvc do beyond removing that file?  It's not clear to me.  I expected `dvc remove` to do the inverse of `dvc add`.  Why doesn't it?  Is there some other way to do the inverse of `dvc add`?
1. In the [user guide](https://dvc.org/doc/user-guide/update-tracked-file), I feel like it should be more clear that this command is destructive.  Due to the misleading syntax issues above, and apparently reading too quickly, I didn't catch until too late that the statement `This will remove train.tsv from the working dir` did not match the file in the command.  Another suggestion might be to put the non-destructive option (`dvc unlink`) first, or clearly label those headings as [Destructive] and [Non-destructive].

**Comment by shcheklein** [comment]
Hi, @colllin. That sounds bad. First of all, let's try to run `dvc checkout some.tgz.dvc`. It should get your file back.

**Comment by shcheklein** [comment]
`dvc remove` is safe in a sense that it does not remove your data from the cache (`dvc add` or `dvc run` puts files into `.dvc/cache`). It removes it _only_ from your _workspace_. It is very similar to git, if you have a code file that is committed into git you can always get it back via `git checkout -- file`. The same here, just run `dvc checkout file.dvc` and it should get it back.

**Comment by shcheklein** [comment]
Sorry for the confusion though. May be we should put a note after `dvc remove`, something like: `Your date is cached, run dvc pull or dvc checkout <file> if you want to get back`. @colllin please, let us know what you think.

**Comment by shcheklein** [comment]
@colllin just want to follow up on this and make sure that you was able to checkout your data back. Please, let us know how things are going.

**Comment by colllin** [comment]
Another comment would be that `dvc unprotect` is unintuitive, too, because I don't remember `dvc protect`ing anything in the first place, so it isn't clear why I would need to undo something I haven't done, especially when it doesn't seem like there's any way to re-protect it afterward, nor do I understand what "protection" I'd be giving up.  It feels like `dvc unprotect` is designed to perform the inverse of `dvc add`, but the word "remove" is the semantic inverse of "add", and there's a `dvc remove` which doesn't seem to perform the inverse of `dvc add`.

**Comment by colllin** [comment]
As we said, I don’t care about tracking changes to my dataset. I’m more interested in the linking between datasets and fit commits, and the `dvc repro` framework for running experiments.   
  

  
That said, sometimes I do need to modify my dataset, for example I discovered some corrupt labels and removed those samples.
  

  
Note that at this point I’m talking about modifying the extracted, uncached dataset — image files and mask files.   
  

  
So then I re-tarred this dataset in the hopes of replacing the original tarball, while keeping the accompanying pipeline which extracts the tarball to a specific location.
  

  
After I had recreated the tarball, it was not clear how to add the new tarball to dvc as a replacement for the original.   
  

  
At this point I tried `dvc remove some.tgz.dvc`, which then deleted the tarball I had just spent hours creating.   
  

  
Is my state and goal more clear now? What was the correct way to accomplish it?
  

  
  

  
  
>

**Comment by shcheklein** [comment]
@colllin I think I have a better sense now. Just to be completely on the same page. 

Let's say you have `images.tgz` and `images.tgz.dvc`, and `images` initially. You made some changes (remove files, add files) inside the `images` directory. And right after that you ran `tar zcf images.tgz image`, basically overwriting the tar-ball. Is it correct?

To go back, and answer your initial questions. The workflow to update the tarball should have looked like this:

```
dvc unprotect images.tgz              # or dvc remove images.tgz.dvc - it's not destructive
<add/remove images or metafiles inside images>
tar zcf images.tgz images              # why do we use -z, is it actually beneficial?
dvc add images.tgz
```

**What does `dvc unprotect` command do?** It unlinks the file in your workspace (the one you actually see, the `images.tgz` in this case) from the cached version (the one inside `.dvc/cache`) to prevent cache corruption in case you actually overwrite the file in the workspace thus overwriting the cached (previous) version along the way. 

**What does `dvc remove smth` command do?** It's a syntax sugar for the `rm`. It's beneficial when you run it with a stage file that have multiple outputs, for example. It also gives value when you use a protected mode by default (when all your files that are under DVC control set to be read only). In this case it's more like `chmod` + `rm`.

From what I see, it's definitely confusing to see that `remove` takes dvc stage files, `unprotect` just files, `move` just files. It's a good suggestion to make them all consistent with each other, accept files and provide an option to accept dvc stage files, for example.

Another thing is that `dvc remove` should warn a user if file is not in cache or remote storage before removing it from the working space.

 Any thoughts, @efiop @dmpetrov @colllin ?

**Comment by colllin** [comment]
@shcheklein Thank you for your time and thorough responses.  ❤️ 

I think the only confusing thing about my situation at the beginning is that I started out with the `tar` command _first_, which, based on my understanding after reading through #1599, means that I had already corrupted my cache.  I think we're already addressing that in #1599.  

As a new user, this `dvc` behavior was unlike git, where it's always safe to modify your files, and then you sort out the intended changes later before committing.  

Feel free to close this issue if it is not helping you track anything.
